src/App.tsx

```diff
@@ -1,8 +1,45 @@
 import './App.css'
+import { useAsyncEffect } from './hooks/useAsyncEffect'
+import {
+  getAccessToken,
+  oauth2Request,
+  redirectToAutorizationPage,
+} from './services/OAUTH2'
+import { redditEndpoints } from './constants'
+import { APPNAME, API, SECRET } from './keys.json'
 
-function App() {
+export const App = () => {
+  // useAsyncEffect(async () => {
+  //   const { access_token } = await oauth2Request(
+  //     redditEndpoints['access'],
+  //     APPNAME,
+  //     API,
+  //     SECRET,
+  //   )
+  //   console.log(access_token)
+  // })
   return (
-   <></>
+    <>
+      <button
+        onClick={() =>
+          redirectToAutorizationPage(API, 'http://localhost:5173/callback')
```

Comment:
This link wouldn't work on hypothetical prod

Reply:
fixed